File: attachment-microservice/src/main/scala/data/service/impl/LocalStorageService.scala
package data.service.impl

import java.io.{BufferedOutputStream, File, FileOutputStream, InputStream}
import java.util.UUID

import data.entity.Attachment
import data.service.StorageService
import org.springframework.stereotype.Service
import org.springframework.util.FileCopyUtils

/**
  * local storage strategy
  *
  * @author 刘佳兴
  * @version V1.0
  */
@Service
class LocalStorageService extends StorageService{

  val storagePath = "/tmp"

  override def write(inputStream: InputStream, attachment: Attachment): Unit = {
    val stream = new BufferedOutputStream(new FileOutputStream(new File(storagePath + attachment.relativePath + attachment.name + attachment.suffix)))
    FileCopyUtils.copy(inputStream, stream)
  }

  override def read(path: String): File = {
    new File(storagePath + path)
  }

}

File: attachment-microservice/src/main/java/data/service/impl/LocalStorageService.java
package data.service.impl;

import data.entity.Attachment;
import data.service.StorageService;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Service;
import org.springframework.util.FileCopyUtils;

import java.io.*;

/**
 * @author 刘佳兴
 * @version V1.0
 */
@Service
public class LocalStorageService implements StorageService {

    @Value("${storage.local.path}")
    String storagePath;
    @Value("${storage.local.url}")
    String url;

    @Override
    public String write(InputStream inputStream, Attachment attachment) throws IOException {
        OutputStream stream = new BufferedOutputStream(new FileOutputStream(new File(storagePath + File.separator + attachment.getRelativePath() + attachment.getName())));
        FileCopyUtils.copy(inputStream, stream);
        return url + attachment.getName();
    }

    @Override
    public File read(String path) {
        return new File(storagePath + path);
    }

}
